history-network.md

```diff
@@ -0,0 +1,300 @@
+
+If the size of a message would exceed the max packet size of 1280 bytes, then the transmission of the message contents would occur over [uTP](https://github.com/ethereum/stateless-ethereum-specs/blob/master/discv5-utp.md).
+
+uTP connections are initiated with randomly generated connection IDs. Upon sending a message with some `connection-id`, the sender should initiate a uTP stream using `connection-id`. Upon receiving a message that contains some `connection-id`, the recipient should listen for an incoming uTP connection with `connection-id`.
+
+NOTE: The `Offer`/`Accept`/`Store` messages do not conform to the request/response paradigm of Discovery v5. We plan to propose a change to the base protocol definition to loosen the language.
+
+#### Ping (0x01)
+
+Communicate `radius` information of the sender, and request `radius` information of the recipient.
+
+```
+protocol-message-type = 0x01
+protocol-message      = Container(radius: uint256)
+```
+
+#### Pong (0x02)
+
+In response to a `Ping` message, communicate `radius` information of the sender.
+
+```
+protocol-message-type = 0x02
+protocol-message      = Container(radius: uint256)
+```
+
+#### FindNode (0x03)
+
+Request a list of ENR values for nodes whose `logdistance` from the recipient's `node-id` is equal to one of the specified values.
+
+```
+protocol-message-type = 0x03
+protocol-message      = Container(distances: List[uint8, 256])
+```
+
+Each element of `distances` **MUST** be unique. A zero value in the `distances` list represents a request for the recipient's ENR value.
+
+#### FoundNodes (0x04)
+
+In response to a `FindNode` message, communicate a list of RLP-encoded ENR values for nodes whose `logdistance` value satisfies the request.
+
+```
+protocol-message-type = 0x04
+protocol-message      = Container(total: uint8, enrs: List[Bytes, 32])
+```
+
+The recipient may know of more than 32 nodes that satisfy the request. In this case, the recipient will need to send multiple `FoundNodes` messages in response to the `FindNode` message.
+
+Here, `total` denotes the total number of `FoundNodes` messages that the sender of the `FindNode` message should expect in response.
+
+Each ENR **MUST** be unique to some `node-id`. Each ENR **MUST** correspond to an element in the `distances` field of the `FindNode` request.
+
+#### FindContent (0x05)
+
+Request the data for `content` with `content-key`, or a list of ENR values for nodes whose `distance` from `content-id` is smaller than that of the recipient.
+
+The encoding of `content-key` for all possible data items is specified above.
+
+```
+protocol-message-type = 0x05
+protocol-message      = Container(content-key: Bytes)
+```
+
+#### FoundContent (0x06)
+
+In response to a `FindContent` message, communicate one of the following:
+
+* A connection ID for a uTP stream to transmit the requested  data
+* A list of RLP-encoded ENR values with maximum length of 32
+* A byte-array that encodes the requested data
```

Comment:
I would add in spec that all history content i.e headers, bodies and transactions should be rlp encoded during wire transfer

Reply:
Yes.  I would suggest we envelope in a SSZ `List[Bytes]` of individual payloads.